net/security-context/tls-stream.cpp

```diff
@@ -76,17 +76,21 @@ class GlobalSSLContext : public Singleton<GlobalSSLContext> {
         OpenSSL_add_all_ciphers();
         OpenSSL_add_all_digests();
         OpenSSL_add_all_algorithms();
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
```

Comment:
Is there a function to obtain openssl version dynamically? If so, it's better to use ```if``` instead of ```#if```.

Reply:
```c
 int SSL_client_version(const SSL *s);

 const char *SSL_get_version(const SSL *ssl);

 int SSL_is_dtls(const SSL *ssl);

 int SSL_version(const SSL *s);
```
Some of the variants seem to return numerical version?

https://www.openssl.org/docs/man1.1.1/man3/SSL_get_version.html

Reply:
According to https://github.com/openssl/openssl/blob/OpenSSL_1_1_0-stable/crypto/cversion.c

we can use ```SSLeay()``` for versions before 1.1, and ```OpenSSL_version_num()``` for those after 1.1.

Reply:
We can probe for the above functions with ```dlopen()``` and ```dlsymbol()```, and decide which one to use.